athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/HbaseRecordHandler.java

```diff
@@ -119,18 +126,41 @@ protected void readWithConstraint(BlockSpiller blockSpiller, ReadRecordsRequest
         String conStr = split.getProperty(HBASE_CONN_STR);
         boolean isNative = projection.getCustomMetadata().get(HBASE_NATIVE_STORAGE_FLAG) != null;
 
-        //setup the scan so that we only read the key range associated with the region represented by our Split.
-        Scan scan = new Scan(split.getProperty(START_KEY_FIELD).getBytes(), split.getProperty(END_KEY_FIELD).getBytes());
-
-        //attempts to push down a partial predicate using HBase Filters
-        scan.setFilter(pushdownPredicate(isNative, request.getConstraints()));
+        String schemaName;
+        String tableName;
+        Scan scan;
+        if (request.getConstraints().isQueryPassThrough()) {
+            Map<String, String> qptArguments = request.getConstraints().getQueryPassthroughArguments();
+            queryPassthrough.verify(qptArguments);
+            schemaName = qptArguments.get(HbaseQueryPassthrough.DATABASE);
+            tableName = qptArguments.get(HbaseQueryPassthrough.COLLECTION);
+            String filter = qptArguments.get(HbaseQueryPassthrough.FILTER);
+            scan = new Scan();
+            if (!StringUtils.isEmpty(filter)) {
```

Comment:
Just adding a comment here to state that "empty filter" is allowed. Otherwise it looks odd as to why we don't have else statement attached to this condition.

Reply:
Hi @AbdulR3hman Added the comment as per your suggestion.